interp/jit_test.go

```
@@ -0,0 +1,800 @@
+// i.fr, or asserting staticPlan's return shape without ever running it. The
+// subset of this contract reachable by warming a program up naturally and
+// observing Run's result or a profiler metric lives in
+// interp_test.TestARM64_StaticPlanMatchesThreaded instead.
+func TestCompiler_Compile(t *testing.T) {
+	if runtime.GOARCH == "arm64" {
+		t.Run("guard value", func(t *testing.T) {
+			prog := program.New([]instr.Instruction{
+				instr.New(instr.GLOBAL_GET, 0), instr.New(instr.GLOBAL_GET, 1), instr.New(instr.I32_DIV_S),
+			}, program.WithGlobals(types.TypeI32, types.TypeI32))
+			i := New(prog, WithThreshold(-1))
+			defer i.Close()
```

golangci-lint (2.12.2)

[error] Error return value of `i.Close` is not checked

(errcheck)